src/components/Post/PostInFeed.tsx

```diff
@@ -1,50 +1,62 @@
 import { Box, Text, Skeleton, HStack, FavouriteIcon, Pressable } from 'native-base';
 import { Post } from '../../xplat/types/post';
 import { useEffect, useState } from 'react';
+import { useQuery } from 'react-query';
 import '../css/feed.css';
 import { User } from '../../xplat/types/user';
 import AuthorHandle from '../User/AuthorHandle';
 import PostImages from './PostImages';
+import { buildPostFetcher } from '../../utils/queries';
 
 const PostInFeed = ({ post, _passAuthor }: { post: Post, _passAuthor?: User | undefined }) => {
     const [body, setBody] = useState('');
     const [author, setAuthor] = useState<User>();
     const [imageCount, setImageCount] = useState(0);
     const [imageURLs, setImageURLs] = useState<Array<string | undefined>>();
-    const [isLoaded, setIsLoaded] = useState(false);
+    const [videoURL, setVideoURL] = useState<string | undefined>();
+    const [videoThumbnail, setVideoThumbnail] = useState<string | undefined>();
+    const [timestamp, setTimestamp] = useState<Date>();
     const [likes, setLikes] = useState(0);
-
-    async function resolvePost (): Promise<void> {
-        await post.getData();
-
-        void post.getTextContent().then(setBody);
-        void post.getImageContentUrls().then(setImageURLs);
-        void post.getImageCount().then(setImageCount);
-        if (_passAuthor)
-            setAuthor(_passAuthor);
-        else
-            void post.getAuthor().then(setAuthor);
-        void post.getLikes().then((likesArray) => { setLikes(likesArray.length); });
-        setIsLoaded(true);
-    }
+    const { isLoading, error, data } = useQuery(
+        post.docRef!.id, 
+        buildPostFetcher(post)   
+    );
 
     useEffect(() => {
-        void resolvePost();
-    }, []);
+        if (data && !error)
```

Comment:
`error` is a string containing the error, `isError` is a boolean that says whether or not we have an error.

Reply:
Oh okay thanks, I must've missed it in the docs